src/app/modules/verification/verification.controller.ts

```diff
@@ -168,6 +175,81 @@ export const handleGetOtp: RequestHandler<
     })
 }
 
+/**
+ * NOTE: This is exported solely for testing
+ * Generates an otp when a user requests to verify a field.
+ * The current answer is signed, and the signature is also saved in the transaction, with the field id as the key.
+ * @param answer The mobile or email number of the user
+ * @param transactionId The id of the transaction to verify
+ * @param formId The id of the form to verify
+ * @param fieldId The id of the field to verify
+ * @returns 201 when otp generated successfully
+ * @returns 400 when the parameters could not be parsed
+ * @returns 400 when the transaction has expired
+ * @returns 400 when the otp could not be sent via sms
+ * @returns 400 when the otp could not be sent via email
+ * @returns 400 when the provided phone number is not valid
+ * @returns 400 when the field type is not supported for validation
+ * @returns 404 when the requested form was not found
+ * @returns 404 when the transaction was not found
+ * @returns 404 when the field was not found
+ * @returns 422 when the user requested for a new otp without waiting
+ * @returns 500 when the otp could not be hashed
+ * @returns 500 when there is a database error
+ */
+export const _handleGenerateOtp: RequestHandler<
+  { transactionId: string; formId: string; fieldId: string },
+  ErrorDto,
+  { answer: string }
+> = async (req, res) => {
+  const { transactionId, formId, fieldId } = req.params
+  const { answer } = req.body
+  const logMeta = {
+    action: 'handleGenerateOtp',
+    transactionId,
+    fieldId,
+    ...createReqMeta(req),
+  }
+  return FormService.retrieveFormById(formId)
```

Comment:
could you add a comment here about why we retrieve the form and then don't do anything with it?